.github/workflows/test_coverage.yaml

```diff
@@ -0,0 +1,41 @@
+name: Unit Test Coverage
+on:
+  push:
+    branches:
+      - master
```

Comment:
main

Are you sure you want this also on pushes, and not just pull requests? One alternative approach is to block direct commits to main, and force everyone to go through pull requests.

Reply:
eventually, the blocking of commits is what I want, but there I need to find a C++ reviewer first.